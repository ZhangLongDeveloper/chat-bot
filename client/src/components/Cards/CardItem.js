import React from 'react';
import PropTypes from 'prop-types';

const CardItem = ({ data }) => {
  const {
    title, image, description, link,
  } = data;

  const renderLink = () => {
    if (link) {
      return (
        <div className="card-action">
          <a href={link}>View</a>
        </div>
      );
    }
    return null;
  };

  return (
    <div className="card-containter">
      <div className="card">
        <div className="card-image">
          <img src={image} alt="card-pic" />
          <span className="card-title">{title}</span>
        </div>
        <div className="card-content">
          <p>{description}</p>
        </div>
        {renderLink()}
      </div>
    </div>
  );
};

CardItem.propTypes = {
  data: PropTypes.object.isRequired,
};

export default CardItem;
