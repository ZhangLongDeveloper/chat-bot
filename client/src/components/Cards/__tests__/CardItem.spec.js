import React from 'react';
import { create } from 'react-test-renderer';

import CardItem from '../../Cards/CardItem';

const mockComponent = props => (
  <CardItem {...props} />
);

describe('components/Cards/<CardItem />', () => {
  it('should render component', () => {
    const props = {
      data: {
        title: 'title',
        image: 'image',
        description: 'description',
        link: 'link',
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component without link', () => {
    const props = {
      data: {
        title: 'title',
        image: 'image',
        description: 'description',
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
