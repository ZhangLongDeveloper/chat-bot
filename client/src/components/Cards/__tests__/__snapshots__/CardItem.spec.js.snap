// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Cards/<CardItem /> should render component 1`] = `
<div
  className="card-containter"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt="card-pic"
        src="image"
      />
      <span
        className="card-title"
      >
        title
      </span>
    </div>
    <div
      className="card-content"
    >
      <p>
        description
      </p>
    </div>
    <div
      className="card-action"
    >
      <a
        href="link"
      >
        View
      </a>
    </div>
  </div>
</div>
`;
